### PR TITLE
Migrate to target specifier supported by release flag in newer JDKs

### PR DESCRIPTION
**Pull request by pzygielo**
This would allow to use `mojo.java.target` directly to configure `maven.compiler.release`.

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
ups:

```
Error:  Rule 4: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
Detected JDK Version: 1.8.0-352 is not in the allowed range 8.
```

Looks like issue/bug in enforcer-maven-plugin

**Comment by pzygielo** [comment]
I'm going push change to `minimalJavaBuildVersion` and add some comment here after that.

**Comment by pzygielo** [comment]
I guess there is only documentation purpose of `requireJavaVersion` used here.

With JDK 1.7 I'm getting:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.1.0:enforce (mojo-enforcer-rules) on project mojo-parent: Execution mojo-enforcer-rules of goal org.apache.maven.plugins:maven-enforcer-plugin:3.1.0:enforce failed: Unable to load the mojo 'enforce' in the plugin 'org.apache.maven.plugins:maven-enforcer-plugin:3.1.0' due to an API incompatibility: org.codehaus.plexus.component.repository.exception.ComponentLookupException: org/apache/maven/plugins/enforcer/EnforceMojo : Unsupported major.minor version 52.0
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.apache.maven.plugins:maven-enforcer-plugin:3.1.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
```
anyway.

**Comment by slawekjaranowski** [comment]
> I guess there is only documentation purpose of `requireJavaVersion` used here.
> 
> With JDK 1.7 I'm getting:
> 
> ```
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.1.0:enforce (mojo-enforcer-rules) on project mojo-parent: Execution mojo-enforcer-rules of goal org.apache.maven.plugins:maven-enforcer-plugin:3.1.0:enforce failed: Unable to load the mojo 'enforce' in the plugin 'org.apache.maven.plugins:maven-enforcer-plugin:3.1.0' due to an API incompatibility: org.codehaus.plexus.component.repository.exception.ComponentLookupException: org/apache/maven/plugins/enforcer/EnforceMojo : Unsupported major.minor version 52.0
> [ERROR] -----------------------------------------------------
> [ERROR] realm =    plugin>org.apache.maven.plugins:maven-enforcer-plugin:3.1.0
> [ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
> ```
> 
> anyway.

No way for 1.7

File: pom.xml [hunk 1]
```diff
@@ -186,11 +186,12 @@
...
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <mojo.java.target>1.8</mojo.java.target>
+    <mojo.java.target>8</mojo.java.target>
     <maven.compiler.source>${mojo.java.target}</maven.compiler.source>
     <maven.compiler.target>${mojo.java.target}</maven.compiler.target>
 
-    <minimalJavaBuildVersion>${mojo.java.target}</minimalJavaBuildVersion>
+    <!-- as expected by Enforcer -->
+    <minimalJavaBuildVersion>1.8</minimalJavaBuildVersion>
```
Comment (slawekjaranowski) [inline pom.xml:194]: It a little will complicate when project set target as 8 also want to check if used JDK is ok. 
In such case we will must remember to set another property also.

My proposition is to postpone this, and allow 8 as 1.8  for maven enforcer.

Comment (slawekjaranowski) [reply pom.xml:194]: https://issues.apache.org/jira/browse/MENFORCER-440

**Comment by pzygielo** [comment]
> No way for 1.7

I don't understand.
I think I've said the same.

My question is - what's the point of requiring Java Version _with enforcer_ to be at least 1.8 when that condition can't be validated with earlier JDK?

That's what bothers me: JDK at least 8 is (currently) _enforced_ by the m-enforcer-p anyway  in not very friendly manner, without the rule itself.

**Comment by slawekjaranowski** [comment]
> 
> My question is - what's the point of requiring Java Version _with enforcer_ to be at least 1.8 when that condition can't be validated with earlier JDK?
> 
> That's what bothers me: JDK at least 8 is (currently) _enforced_ by the m-enforcer-p anyway in not very friendly manner, without the rule itself.

Similar:
https://issues.apache.org/jira/browse/MNG-7566

Currently plugin minimum JDK is not verified ... it cause a error in runtime.

Most of maven core plugins requires 1.8 today, Maven from 3.9 also will required 1.8

By the way, interesting discussion how to inform users of minimum JDK, when user try to execute Maven build with not supported JDK.

File: pom.xml (hunk 1 above)
Comment (slawekjaranowski) [reply pom.xml:194]: enforcer released and upgraded here, we can continue with it

**Comment by slawekjaranowski** [comment]
Suppressed by #359 
@pzygielo thanks for idea

**Pull request closed without merging**